README.md

```diff
@@ -11,6 +11,18 @@ This tool is currently in beta, while any issues are worked through. Please feel
 
 This is intended to be used as a stop-gap for teams who are using Kotlin in a legacy system that started out in Java. New projects or projects that have the ability to migrate to newer tools made for Kotlin from the ground up should do so. [Kotlin Serialization](https://github.com/Kotlin/kotlinx.serialization) and [Moshi](https://github.com/square/moshi) are great examples of these tools.
 
+## Disclaimer
+This library uses the Kotlin Reflection library and can incur significant performance losses both when constructing the adapter and when converting the JSON to a model. This incurs a large penalty on performance and can adversely affect your application. It is highly suggested that performance tests be done before using this library to ensure that the performance losses are acceptable for your specific application.
+
+Both [Kotlin Serialization](https://github.com/Kotlin/kotlinx.serialization) and [Moshi](https://github.com/square/moshi) support code generation and are much more efficient for this reason.
+
+The adapter is designed to front-load as much of the heavy workload as possible, so using cache warming techniques can make significant improvements.
+```kotlin
+    private fun warmClassCaches(vararg classes: Class<*>) {
+        classes.forEach { gson.getAdapter(it) }
+    }
```

Comment:
Is there any reason not to just provide a method like this from the library?

Reply:
Nope, we can add it.